### DISCUSSION TITLE
Extending JsonReporter

### DISCUSSION THREAD

**Discussion by joemull** [vitest-dev/vitest#3937] [Q&A] [answered]
What import path would I use to import JsonReporter if I wanted to extend it?

I would like to create a custom reporter that just overrides the `writeReport` method to add a few things to meet this spec: https://gradescope-autograders.readthedocs.io/en/latest/specs/   Here's how it might work:

```
function modifyReport (report) {
  const modifiedReport = report
  return modifiedReport
}

export class GradescopeReporter extends JsonReporter {
  async writeReport (report) {
    super.writeReport(modifyReport(report))
  }
}
```

I just can't figure out what import statement to use to get the JsonReporter class. Is it possible or would I have to redefine all the methods, starting by extending the Reporter class? What am I missing?

**Comment by sheremet-va** [comment]
I'm not sure if we publicly export it anywhere to be honest

**Reply by joemull** [reply]
Ok, thanks. Would it meet your design goals to do so? I'm happy to open in issue if it's something you'd consider.

**Reply by sheremet-va** [reply, marked as answer]
Sure, I'm not against exporting all reporters

**Reply by joemull** [reply]
OK, thanks, I've opened an issue
